smile_id_core/Signature.py

```diff
@@ -17,42 +17,16 @@ def __init__(self, partner_id: str, api_key: str):
         self.partner_id = partner_id
         self.api_key = api_key
 
-    def generate_sec_key(self, timestamp=None):
-        public_key = RSA.importKey(base64.b64decode(self.api_key))
-        cipher = PKCS1_v1_5.new(public_key)
-        if timestamp is None:
-            timestamp = int(time.time())
-        hashed = self.__get_hash(timestamp)
-        encrypted = base64.b64encode(cipher.encrypt(hashed.encode("utf-8")))
-
-        signature = f"{encrypted.decode(encoding='UTF-8')}|{hashed}"
-        return {"sec_key": signature, "timestamp": timestamp}
-
-    def __get_hash(self, timestamp):
-        to_hash = f"{int(self.partner_id)}:{timestamp}"
-        new_hash = str(to_hash).encode("utf-8")
-        return hashlib.sha256(new_hash).hexdigest()
-
-    def generate_signature(self, timestamp=None):
-        _timestamp = timestamp
-        if _timestamp is None:
-            _timestamp = datetime.now().isoformat()
+    def generate_signature(self, timestamp=datetime.now().isoformat()):
         hmac_new = hmac.new(self.api_key.encode("utf-8"), digestmod=hashlib.sha256)
-        hmac_new.update(_timestamp.encode("utf-8"))
+        hmac_new.update(timestamp.encode("utf-8"))
         hmac_new.update(str(self.partner_id).encode("utf-8"))
         hmac_new.update("sid_request".encode("utf-8"))
         calculated_signature = base64.b64encode(hmac_new.digest())
         return {
             "signature": calculated_signature.decode("utf-8"),
-            "timestamp": _timestamp,
+            "timestamp": timestamp,
         }
 
-    def confirm_signature(self, timestamp, msg_signature):
+    def confirm_signature(self, timestamp: str, msg_signature):
```

Comment:
Can you complete the type signature for the entire function? i.e. provide a type for `msg_signature` and for the return value of the function

Reply:
@ayinloya This one seems unresolved to me

I think it should be something like this

```suggestion
    def confirm_signature(self, timestamp: str, msg_signature: str) -> bool:
```